pkg/controller/runtimecomponent/runtimecomponent_controller_test.go

```diff
@@ -190,80 +206,187 @@ func TestRuntimeController(t *testing.T) {
-		t.Fatalf("Get runtimecomponent: (%v)", err)
+	// Check that the default service account was deleted
+	if err = r.GetClient().Get(context.TODO(), req.NamespacedName, serviceaccount); err == nil {
+		return err
 	}
+	return nil
+}
+
+func testServiceMonitoring(t *testing.T, r *ReconcileRuntimeComponent, rb appstacksutils.ReconcilerBase) error {
+	runtimecomponent, req := makeRuntimeAndReq()
 
-	// Update runtimecomponentto ensure it requeues
-	runtimecomponent.SetGeneration(1)
+	// Test with monitoring specified
+	runtimecomponent.Spec.Monitoring = &appstacksv1beta1.RuntimeComponentMonitoring{}
 	updateRuntimeComponent(r, runtimecomponent, t)
+	res, err := r.Reconcile(req)
+	if err = verifyReconcile(res, err); err != nil {
+		return err
+	}
 
+	svc := &corev1.Service{ObjectMeta: defaultMeta}
+	if err = r.GetClient().Get(context.TODO(), req.NamespacedName, svc); err != nil {
+		return err
+	}
+
+	monitorTests := []Test{
+		{"Monitor label assigned", "true", svc.Labels["monitor."+runtimecomponent.GetGroupName()+"/enabled"]},
+	}
+	if err = verifyTests(monitorTests); err != nil {
+		return err
+	}
+
+	// Test without monitoring on
+	runtimecomponent.Spec.Monitoring = nil
+	updateRuntimeComponent(r, runtimecomponent, t)
 	res, err = r.Reconcile(req)
-	if err != nil {
-		t.Fatalf("reconcile: (%v)", err)
+	if err = verifyReconcile(res, err); err != nil {
+		return err
+	}
+
+	svc = &corev1.Service{ObjectMeta: defaultMeta}
+	if err = r.GetClient().Get(context.TODO(), req.NamespacedName, svc); err != nil {
+		return err
+	}
+
+	monitorTests = []Test{
+		{"Monitor label unassigned", "", svc.Labels["app."+runtimecomponent.GetGroupName()+"/monitor"]},
+	}
+	if err = verifyTests(monitorTests); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addResourcesToScheme(t *testing.T, s *runtime.Scheme, runtimecomponent *appstacksv1beta1.RuntimeComponent) {
+	if err := servingv1alpha1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add servingv1alpha1 scheme: (%v)", err)
+	}
+
+	if err := routev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme: (%v)", err)
+	}
+
+	if err := imagev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add image scheme: (%v)", err)
+	}
+
+	if err := applicationsv1beta1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add image scheme: (%v)", err)
+	}
+
+	if err := certmngrv1alpha2.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add cert-manager scheme: (%v)", err)
+	}
+
+	if err := prometheusv1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add prometheus scheme: (%v)", err)
 	}
+
+	s.AddKnownTypes(appstacksv1beta1.SchemeGroupVersion, runtimecomponent)
+	s.AddKnownTypes(certmngrv1alpha2.SchemeGroupVersion, &certmngrv1alpha2.Certificate{})
+	s.AddKnownTypes(prometheusv1.SchemeGroupVersion, &prometheusv1.ServiceMonitor{})
 }
 
 // Helper Functions
+func makeRuntimeAndReq() (*appstacksv1beta1.RuntimeComponent, reconcile.Request) {
```

Comment:
This helper function groups two unrelated functions, which usually isn't the best practice even if it reduces duplication a bit.  Looking at it now the `reconcile.Request` returned by `createReconcileRequest` is always the same, it could instead just be a global variable initialized at the top with the others.  You could then remove this function as well as `createReconcileRequest`